src/compiler/sourcemapDecoder.ts

```diff
@@ -61,7 +61,7 @@ namespace ts.sourcemaps {
 
     export function decode(host: SourceMapDecodeHost, mapPath: string, map: SourceMapData, program?: Program, fallbackCache = createSourceFileLikeCache(host)): SourceMapper {
         const currentDirectory = getDirectoryPath(mapPath);
-        const sourceRoot = map.sourceRoot || currentDirectory;
+        const sourceRoot = map.sourceRoot ? getNormalizedAbsolutePath(map.sourceRoot, currentDirectory) : currentDirectory;

```

Comment:
Do you need to handle if the sourceRoot specified is URL ?

Reply:
`getNormalizedAbsolutePath` already passes along URLs just fine (it preserves the input root if it is rooted), as will the rest of the path handling in the sourcemap code. The language service will likely summarily reject the mapped location, however, since `https://github.com/Microsoft/TypeScript/tree/master/src/compiler/checker.ts` (for example) won't exist on disk (though I imagine a host implementation could exist that returns results for such paths). UNC paths might in theory work fine, though.... those return normal results via `fileExists` and similar APIs on windows, right?

In any case: No, I don't think so - all the path normalizing/building stuff already is.